src/EventLoop/Driver/StreamSelectDriver.php

```diff
@@ -166,12 +166,17 @@ protected function activate(array $callbacks): void
                 $this->timerQueue->insert($callback);
             } elseif ($callback instanceof SignalCallback) {
                 if (!isset($this->signalCallbacks[$callback->signal])) {
-                    if (!@\pcntl_signal($callback->signal, \Closure::fromCallable([$this, 'handleSignal']))) {
-                        $message = "Failed to register signal handler";
-                        if ($error = \error_get_last()) {
-                            $message .= \sprintf("; Errno: %d; %s", $error["type"], $error["message"]);
-                        }
-                        throw new \Error($message);
+                    \set_error_handler(static function (int $errno, string $errstr): bool {
+                        throw new \Error(\sprintf("Failed to register signal handler; Errno: %d; %s", $errno, $errstr));
```

Comment:
Should this be a `RuntimeException` instead?

Reply:
yes, it's not an Error.

Reply:
Probably our unsupported feature exception could be used?